### PR TITLE
[Chip-tool] Not register LIT at default

### PR DESCRIPTION
**Pull request by yunhanw-google**
Per https://github.com/project-chip/connectedhomeip/issues/31992, at default, chip-tool would register LIT unless skip-icd-registration is false.

### CODE REVIEW

**Review by mkardous-silabs — APPROVED** [review]

**Review by bauerschwan — APPROVED** [review]

**Comment by andy31415** [comment]
@yunhanw-google both darwin and linux seem to fully hang in tests ... I believe this may be something to do with the PR (although I am not sure what)

**Review by andy31415 — DISMISSED** [review]
Requesting changes as a marker that unit tests seem to hang. This does not look like a flake.

File: scripts/tests/chiptest/test_definition.py [hunk 1]
```diff
@@ -372,7 +372,8 @@ def Run(self, runner, apps_register, paths: ApplicationPaths, pics_file: str,
                     runner.RunSubprocess(python_cmd, name='CHIP_REPL_YAML_TESTER',
                                          dependencies=[apps_register], timeout_seconds=timeout_seconds)
             else:
-                pairing_cmd = paths.chip_tool_with_python_cmd + ['pairing', 'code', TEST_NODE_ID, setupCode]
+                pairing_cmd = paths.chip_tool_with_python_cmd + \
+                    ['pairing', 'code', TEST_NODE_ID, setupCode, 'skip-icd-registration', '0']
```
Comment (mkardous-silabs) [inline scripts/tests/chiptest/test_definition.py:376]: Now that i think about this, should we be adding this?

Comment (yunhanw-google) [reply scripts/tests/chiptest/test_definition.py:376]: for manual test, we just provide the convenience to disable icd registration at default when using chip-tool based upon the ticket during certification test,
https://github.com/project-chip/connectedhomeip/issues/31992, However, we need exercise the ICD registration in auto-commissioning flow in CI, which is the real thing darwin/android platform would do in production.

Comment (mkardous-silabs) [reply scripts/tests/chiptest/test_definition.py]: Does this affect all automated tests in CI? Test plan assumption is that chip-tool will not register by default.

Comment (yunhanw-google) [reply scripts/tests/chiptest/test_definition.py]: it only impacts the test cases when chip-tool validate with lit-example-app

Comment (yunhanw-google) [reply scripts/tests/chiptest/test_definition.py]: offline resolved.

File: scripts/tests/chiptest/test_definition.py [hunk 2]
```diff
@@ -372,7 +372,8 @@ def Run(self, runner, apps_register, paths: ApplicationPaths, pics_file: str,
                     runner.RunSubprocess(python_cmd, name='CHIP_REPL_YAML_TESTER',
                                          dependencies=[apps_register], timeout_seconds=timeout_seconds)
             else:
-                pairing_cmd = paths.chip_tool_with_python_cmd + ['pairing', 'code', TEST_NODE_ID, setupCode]
+                pairing_cmd = paths.chip_tool_with_python_cmd + \
+                    ['pairing', 'code', TEST_NODE_ID, setupCode, '--skip-icd-registration', '0']
```
Comment (bzbarsky-apple) [inline scripts/tests/chiptest/test_definition.py:376]: Should have `'false'`, not `'0'`....

**Review by bzbarsky-apple — COMMENTED** [review]
But also, should we make it a "register as ICD client" boolean (defaulting to false) or something?  The multiple negatives are a bit hard to follow.

File: scripts/tests/chiptest/test_definition.py [hunk 3]
```diff
@@ -372,7 +372,8 @@ def Run(self, runner, apps_register, paths: ApplicationPaths, pics_file: str,
                     runner.RunSubprocess(python_cmd, name='CHIP_REPL_YAML_TESTER',
                                          dependencies=[apps_register], timeout_seconds=timeout_seconds)
             else:
-                pairing_cmd = paths.chip_tool_with_python_cmd + ['pairing', 'code', TEST_NODE_ID, setupCode]
+                pairing_cmd = paths.chip_tool_with_python_cmd + \
+                    ['pairing', 'code', TEST_NODE_ID, setupCode, '--icd-registration', 'false']
```
Comment (mkardous-silabs) [inline scripts/tests/chiptest/test_definition.py:376]:
```suggestion
                    ['pairing', 'code', TEST_NODE_ID, setupCode, '--icd-registration', 'true']
```

if am not mistaken?

File: examples/chip-tool/commands/pairing/PairingCommand.h [hunk 4]
```diff
@@ -65,8 +65,8 @@ class PairingCommand : public CHIPCommand,
                     "Bypass the attestation verifier. If not provided or false, the attestation verifier is not bypassed."
                     " If true, the commissioning will continue in case of attestation verification failure.");
         AddArgument("case-auth-tags", 1, UINT32_MAX, &mCASEAuthTags, "The CATs to be encoded in the NOC sent to the commissionee");
-        AddArgument("skip-icd-registration", 0, 1, &mSkipICDRegistration,
-                    "Skip registering for check-ins from ICDs during commissioning. Default: false");
+        AddArgument("icd-registration", 0, 1, &mICDRegistration,
+                    "registering for check-ins from ICDs during commissioning. Default: false");
```
Comment (bzbarsky-apple) [inline examples/chip-tool/commands/pairing/PairingCommand.h:69]:
```suggestion
                    "Whether to register for check-ins from ICDs during commissioning. Default: false");
```

**Review by bzbarsky-apple — APPROVED** [review]
Approved with @mkardous-silabs's change made.

File: .github/workflows/tests.yaml [hunk 5]
```diff
@@ -386,7 +386,6 @@ jobs:
                      --ota-requestor-app ./out/darwin-x64-ota-requestor-${BUILD_VARIANT}/chip-ota-requestor-app \
                      --tv-app ./out/darwin-x64-tv-app-${BUILD_VARIANT}/chip-tv-app \
                      --bridge-app ./out/darwin-x64-bridge-${BUILD_VARIANT}/chip-bridge-app \
-                     --lit-icd-app ./out/darwin-x64-lit-icd-${BUILD_VARIANT}/lit-icd-app \
```
Comment (mkardous-silabs) [inline .github/workflows/tests.yaml:389]: Not sure i understand the build removal?

Comment (yunhanw-google) [reply .github/workflows/tests.yaml:389]: test_definition.py applies in both darwin and linux chip-tool, I have not yet figured out way to check which tool is used in test_definition.py via os. We have not yet add LIT support in darwin chip-tool so that it complains icd-registration option cannot be found, we should add this back when we have lit support in darwin chip-tool.
cc @bzbarsky-apple for darwin support

Comment (bzbarsky-apple) [reply .github/workflows/tests.yaml:389]: I don't follow.  We have three possible clients here:

1. darwin-framework-tool
2. chip-tool on Linux
3. chip-tool on Darwin 

Item 1 does not support the LIT ICD bits, and has TestIcdManagementCluster turned off when running with it as client.

Items 2 and 3 support the ICD stuff fine, no?  It's the same code.

If you need to tell apart 1 and 2/3 in test_definition.py, check for `test_runtime == TestRunTime.DARWIN_FRAMEWORK_TOOL_PYTHON` (or !=, depending on what you want).

But you should not need any changes to `.github/workflows/tests.yaml` here.

Comment (yunhanw-google) [reply .github/workflows/tests.yaml:389]: yes, the below is what i have seen, I wanna disable the test for darwinframeworktool

'/Users/runner/work/connectedhomeip/connectedhomeip/scripts/tests/yaml/darwinframeworktool.py', 'pairing', 'code', '0x12344321', 'MT:-24J0AFN00KA0648G00', '--icd-registration', 'true', '--server_path', './out/darwin-x64-darwin-framework-tool-no-ble/darwin-framework-tool', '--server_arguments', 'interactive server --storage-directory /var/folders/bc/7ycrpl196j7dsckgtn2m6g800000gn/T/tmp5g7y6x7b']

Comment (yunhanw-google) [reply .github/workflows/tests.yaml]: updated, thanks

File: scripts/tests/chiptest/test_definition.py [hunk 6]
```diff
@@ -372,7 +372,10 @@ def Run(self, runner, apps_register, paths: ApplicationPaths, pics_file: str,
                     runner.RunSubprocess(python_cmd, name='CHIP_REPL_YAML_TESTER',
                                          dependencies=[apps_register], timeout_seconds=timeout_seconds)
             else:
-                pairing_cmd = paths.chip_tool_with_python_cmd + ['pairing', 'code', TEST_NODE_ID, setupCode]
+                pairing_cmd = paths.chip_tool_with_python_cmd + \
+                    ['pairing', 'code', TEST_NODE_ID, setupCode, '--icd-registration', 'true']
```
Comment (bzbarsky-apple) [inline scripts/tests/chiptest/test_definition.py:376]: This change should be undone, no?

File: scripts/tests/chiptest/test_definition.py [hunk 7]
```diff
@@ -372,7 +372,10 @@ def Run(self, runner, apps_register, paths: ApplicationPaths, pics_file: str,
...
+                if self.target == TestTarget.LIT_ICD and not TestRunTime.DARWIN_FRAMEWORK_TOOL_PYTHON:
```
Comment (bzbarsky-apple) [inline scripts/tests/chiptest/test_definition.py:377]: This check makes no sense.  `TestRunTime.DARWIN_FRAMEWORK_TOOL_PYTHON` is not a boolean.

File: scripts/tests/chiptest/test_definition.py [hunk 8]
```diff
@@ -372,7 +372,10 @@ def Run(self, runner, apps_register, paths: ApplicationPaths, pics_file: str,
                     runner.RunSubprocess(python_cmd, name='CHIP_REPL_YAML_TESTER',
                                          dependencies=[apps_register], timeout_seconds=timeout_seconds)
             else:
-                pairing_cmd = paths.chip_tool_with_python_cmd + ['pairing', 'code', TEST_NODE_ID, setupCode]
+                pairing_cmd = paths.chip_tool_with_python_cmd + \
+                    ['pairing', 'code', TEST_NODE_ID, setupCode]
```
Comment (bzbarsky-apple) [inline scripts/tests/chiptest/test_definition.py:376]: Why is this line being changed at all?

Comment (yunhanw-google) [reply scripts/tests/chiptest/test_definition.py]: updated, thanks

File: scripts/tests/chiptest/test_definition.py (hunk 7 above)
Comment (yunhanw-google) [reply scripts/tests/chiptest/test_definition.py]: updated, thanks

File: scripts/tests/chiptest/test_definition.py (hunk 6 above)
Comment (yunhanw-google) [reply scripts/tests/chiptest/test_definition.py]: updated, thanks
